Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -142,6 +144,22 @@ - (void)webViewWebContentProcessDidTerminate:(WKWebView *)webView
     m_delegate = [[_WKWebExtensionContextDelegate alloc] initWithWebExtensionContext:*this];
 }
 
+WebExtensionContext::WebProcessProxySet WebExtensionContext::processes(WebExtensionEventListenerType type, WebExtensionContentWorldType contentWorldType) const
+{
+    auto pagesEntry = m_eventListenerPages.find({ type, contentWorldType });
+    if (pagesEntry == m_eventListenerPages.end())
+        return { };
+
+    WebProcessProxySet result;
+    for (auto entry : pagesEntry->value) {
+        Ref process = entry.key.process();
```

Comment:
So any active WeakPtr for a page is elevated to a Ref-counted thing to keep it alive. That sounds reasonable, but I wanted to double-check that the goal here is to hold the page "alive" until the extension is finished with it?

Reply:
Oh, I see -- this was just code that moved from `WebExtensionContext::processes`. But I do wonder about the ownership change. Seems like it's intentional, so nothing to do.

Reply:
Yeah this is moved code. The `Ref` thing was recently done by @cdumez. We only need to keep it around to send the message. it is fine if it goes away after that or even before a reply is sent.

Reply:
I actually didn't need to move this, I originally had added some ObjC code here but removed it. I will just move it back and no change this.